### PR TITLE
[ID-111] Add Options endpoint to Ga4GH OpenAPI YAML

### PR DESCRIPTION
**Pull request by Shakespeared**
Adds support for the Options endpoint request as per the spec found [here](https://editor.swagger.io/?url=https://ga4gh.github.io/data-repository-service-schemas/preview/develop/openapi.yaml). 

[Ticket](https://broadworkbench.atlassian.net/browse/ID-111)

### CODE REVIEW

**Review by tlangs — APPROVED** [review]

**Comment by andy7i** [comment]
don't forget to add the ticket # to the title! (and it helps me as a reviewer if it's in the ticket description too!)

**Comment by Shakespeared** [comment]
@andy7i Yep yep yep sorry! Updated with the ticket number, thanks for the reminder.

File: service/src/main/resources/vendor/data-repository-service-1.2.0.yaml [hunk 1]
```diff
@@ -404,7 +404,25 @@ tags:
       ## Authentication
 
 
-      ### BasicAuth
+      ### Discovery
+
+
+      The APIs to fetch [DrsObjects](l#tag/DrsObjectModel) and
```
Comment (andy7i) [inline service/src/main/resources/vendor/data-repository-service-1.2.0.yaml:410]: there's an `l` in front of the `#` in some of the links here. What does that do? (I'm not sure how to test this out ;  I was able to pull up the swagger UI in intellij but most/all(?) links on this page only open blank pages)

File: service/src/main/resources/vendor/data-repository-service-1.2.0.yaml [hunk 2]
```diff
@@ -404,7 +404,25 @@ tags:
...
+      [AccessURLs](#tag/AccessURLModel) may require authorization. The
+      authorization mode may vary between DRS objects hosted by a service. The
+      authorization mode may vary between the APIs to fetch a
+      [DrsObject](l#tag/DrsObjectModel) and an associated
+      [AccessURL](#tag/AccessURLModel). Implementers should indicate how to
+      authenticate to fetch a [DrsObject](l#tag/DrsObjectModel) by implementing
+      the [OptionsOjbect](#operation/OptionsObject) API. Implementers should
```
Comment (andy7i) [inline service/src/main/resources/vendor/data-repository-service-1.2.0.yaml:417]:
```suggestion
      the [OptionsObject](#operation/OptionsObject) API. Implementers should
```

File: service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java [hunk 3]
```diff
@@ -30,6 +30,7 @@
 import io.github.ga4gh.drs.model.AccessMethod;
 import io.github.ga4gh.drs.model.AccessMethod.TypeEnum;
 import io.github.ga4gh.drs.model.AccessURL;
+import io.github.ga4gh.drs.model.AllOfAccessMethodAccessUrl;
```
Comment (andy7i) [inline service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java:33]: I was wondering what this was / why we need it. ( my env can't find this -- I pulled the branch and reloaded Gradle and it comes up as unresolveable for me :/ )

**Review by andy7i — COMMENTED** [review]
looks like it is following the spec as far as I can tell! haven't tested it.

File: service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java (hunk 3 above)
Comment (tlangs) [reply service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java:33]: try a `~/.gradlew build`

**Comment by tlangs** [comment]
@andy7i since this is a spec downloaded from GA4GH, we should probably refrain from fixing typos or changing it in any way. We want to generate exactly what the spec says.

**Comment by andy7i** [comment]
@tlangs oh right.. I forgot about that and was thinking about how much of a coincidence it was that I saw "ojbect" twice recently

File: service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java (hunk 3 above)
Comment (Shakespeared) [reply service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java:33]: It's in order to fix a lint error in the older spec. If you want both a $ref and a description you have to wrap it in `AllOf` which unfortunately then gets propagated through the autogenerated code.

File: service/src/main/resources/vendor/data-repository-service-1.2.0.yaml (hunk 1 above)
Comment (Shakespeared) [reply service/src/main/resources/vendor/data-repository-service-1.2.0.yaml:410]: The spec has a lot of typos 🥲

File: service/src/main/resources/vendor/data-repository-service-1.2.0.yaml (hunk 2 above)
Comment (Shakespeared) [reply service/src/main/resources/vendor/data-repository-service-1.2.0.yaml:417]: > @tlangs oh right.. I forgot about that and was thinking about how much of a coincidence it was that I saw "ojbect" twice recently

It's taking all my self control not to correct the typos

Comment (tlangs) [reply service/src/main/resources/vendor/data-repository-service-1.2.0.yaml:417]: You _could_ make a PR into their repo... https://github.com/ga4gh/data-repository-service-schemas

File: service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java (hunk 3 above)
Comment (andy7i) [reply service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java:33]: gotcha, thanks!!

**Review by andy7i — APPROVED** [review]
LGTM! those typos though 🤮

File: service/src/main/resources/vendor/data-repository-service-1.2.0.yaml (hunk 2 above)
Comment (Shakespeared) [reply service/src/main/resources/vendor/data-repository-service-1.2.0.yaml:417]: I'm in ur repo, correcting ur typos…. 😅

**Pull request merged**